main.go

```diff
@@ -149,21 +149,18 @@ func getConnWithRetry(flags *flagStorage) (c *gcsx.Connection, err error) {
 }
 
 func createStorageHandle(flags *flagStorage) (storageHandle storage.StorageHandle, err error) {
-	tokenSrc, err := auth.GetTokenSource(context.Background(), flags.KeyFile, flags.TokenUrl, true)
```

Comment:
just curious why so much of refactoring being done just to construct auth?

Reply:
Not only just to construct auth:
1. We need `IsProdEndPoint` check while creating Auth token as well as setting WithEndpoint() option. One of the motivation for this refactoring is just to keep this logic at same place.
2. The size of `main.go` was growing if we would kept all these logic in main.go file. Hence, move it together to storageutil package.
3. Also, we will support grpc client. I did all these refactoring in keeping those changes also in mind. And created a separate file for client creation in storageutil.